refinery/core/api.py

```diff
@@ -2018,6 +2019,14 @@ def obj_create(self, bundle, **kwargs):
         user = bundle.request.user
         data = json.loads(bundle.request.body)
         new_ext_group = ExtendedGroup(name=data['name'])
+
+        new_ext_group.name = new_ext_group.name.strip()
+        try:
+            new_ext_group.full_clean()
+        except ValidationError as e:
+            raise ImmediateHttpResponse(HttpBadRequest(
+                'Invalid group creation request: %s.' % e
+            ))
         new_ext_group.save()
```

Comment:
While save() does not perform validation checks, it can still raise IntegrityError for duplicate names, so that exception should be handled here.